dotcom-rendering/src/components/FollowButtons.tsx

```diff
@@ -0,0 +1,127 @@
+import { css } from '@emotion/react';
+import type { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+import { space, textSans } from '@guardian/source-foundations';
+import {
+	SvgCheckmark,
+	SvgNotificationsOff,
+	SvgNotificationsOn,
+	SvgPlus,
+} from '@guardian/source-react-components';
+import { palette } from '../palette';
+
+type IconProps = {
+	isFollowing?: boolean;
+	iconIsFollowing: EmotionJSX.Element;
+	iconIsNotFollowing: EmotionJSX.Element;
+};
+
+const FollowIcon = ({
+	isFollowing,
+	iconIsFollowing,
+	iconIsNotFollowing,
+}: IconProps) => (
+	<div
+		css={css`
+			display: flex;
+			margin: 0;
+			margin-right: ${space[1]}px;
+
+			border-radius: 100%;
+			border: 1px solid ${palette('--follow-icon-fill')};
+
+			svg {
+				margin: 1px;
+			}
+		`}
+		style={{
+			backgroundColor: isFollowing
+				? palette('--follow-icon-fill')
+				: undefined,
+			fill: isFollowing
+				? palette('--follow-icon-background')
+				: palette('--follow-icon-fill'),
+		}}
+	>
+		{isFollowing ? iconIsFollowing : iconIsNotFollowing}
+	</div>
+);
+
+const buttonStyles = css`
+	${textSans.small()}
+	color: ${palette('--follow-text')};
+	background: none;
+	border: none;
+	display: block;
+	margin-top: ${space[1]}px;
+	margin-left: 0;
+	min-height: ${space[6]}px;
+	padding: 0;
+	text-align: left;
+`;
+
+const containerStyles = css`
+	display: flex;
+	align-items: center;
+	column-gap: 0.2em;
+`;
+
+const notificationsTextSpan = ({
+	isFollowing,
+}: Pick<ButtonProps, 'isFollowing'>) => (
+	<span>Notifications {isFollowing ? 'on' : 'off'}</span>
+);
+
+const tagTextSpan = ({
+	isFollowing,
+	displayName,
+}: {
+	isFollowing: boolean;
+	displayName?: string;
+}) => (
+	<span>
+		{isFollowing ? 'Following' : 'Follow'} {displayName}
+	</span>
+);
+
+// bridget props
+type ButtonProps = {
+	isFollowing: boolean;
+	onClickHandler: () => void;
+};
+
+export const FollowNotificationsButton = ({
+	isFollowing,
+	onClickHandler,
+}: ButtonProps) => {
+	return (
+		<button onClick={onClickHandler} type="button" css={buttonStyles}>
+			<span css={containerStyles}>
+				<FollowIcon
+					isFollowing={isFollowing}
+					iconIsFollowing={<SvgNotificationsOn size="xsmall" />}
+					iconIsNotFollowing={<SvgNotificationsOff size="xsmall" />}
+				/>
```

Comment:
[React favours composition over inheritance](https://legacy.reactjs.org/docs/composition-vs-inheritance.html), so you might want inline FollowIcon here.